klient/conf/config.go

```diff
@@ -38,6 +38,11 @@ func New(fileName string) (*rest.Config, error) {
 		return rest.InClusterConfig()
 	}
 
+	// set the desired kube context
+	if kubeContext := ResolveClusterContext(); kubeContext != "" {
```

Comment:
This check should also check to see if kubeConfig is available (either via the --kubeconfig flag value or via [`ResolveKubeConfigFile`](https://github.com/kubernetes-sigs/e2e-framework/blob/db4e9fe7e23d2602c34d1fe3853625e754b1fa41/klient/conf/config.go#L67). If the value of KubeConfig is not provided, this should error out. See comment below for detail.

Reply:
> If the value of KubeConfig is not provided, this should error out.

But if we are not passing a kubeconfig we should hit https://github.com/kubernetes-sigs/e2e-framework/blob/db4e9fe7e23d2602c34d1fe3853625e754b1fa41/klient/conf/config.go#L37 because we're assuming in-cluster-config

My understanding is that we call `ResolveKubeConfigFile` in https://github.com/kubernetes-sigs/e2e-framework/blob/db4e9fe7e23d2602c34d1fe3853625e754b1fa41/examples/real_cluster/main_test.go#L38 and then we construct the config from that file, without using any flag.

I think that you're saying that we could use `ResolveKubeConfigFile` to automatically get the kubeconfig *and* we could use the flags to also do additional configurations, am I correct? But `NewFromFlags()` doesn't take any argument, so I think I'm missing something

Reply:
Hi @maruina . You are correct in your last thinking. 
We could force --kubeconfig to be provided when --context is used or (as i tried to suggest)
* Check for value of kubeConfig
* if empty, call ResolveKubeConfigFile
* if still empty, then error out

Also, if there is a better place to make that check work, you can put it where it works best.

Reply:
Ok, I think I've done that. Please have a look :)